client/src/modules/stock/exit/exit.js

```diff
@@ -117,16 +126,75 @@ function StockExitController(
         headerCellFilter : 'translate',
         cellTemplate : 'modules/stock/exit/templates/expiration.tmpl.html',
       },
-      { field : 'actions', width : 25, cellTemplate : 'modules/stock/exit/templates/actions.tmpl.html' },
+      {
+        field : 'actions',
+        width : 25,
+        cellTemplate : 'modules/stock/exit/templates/actions.tmpl.html',
+      },
     ],
     data : vm.stockForm.store.data,
     fastWatch : true,
     flatEntityAccess : true,
+    showGridFooter : true,
+    gridFooterTemplate,
+    onRegisterApi,
   };
 
   // exposing the grid options to the view
   vm.gridOptions = gridOptions;
 
+  const exportation = new GridExportService(vm.gridOptions);
+
+  /**
+   * @method exportGrid
+   * @description export the content of the grid to csv.
+   */
+  vm.exportGrid = () => {
+    exportation.exportToCsv('Stock_Exit_', formatExportColumns, formatExportRows);
+  };
+
+  /**
+   * @function formatExportColumns
+   *
+   * @description this function will be apply to grid columns as filter for getting new columns
+   *
+   * @param {array} columns - refer to the grid columns array
+   * @return {array} - return an array of column object in this format : { displayName : ... }
+   */
+  function formatExportColumns(columns) {
+    return (columns || [])
+      .filter(col => col.displayName && col.displayName.length)
+      .map(col => ({ displayName : $translate.instant(col.displayName), width : col.width }));
+  }
+
+  /**
+   * @function formatExportRows
+   *
+   * @description this function will be apply to grid columns as filter for getting new columns
+   *
+   * @param {array} rows - refer to the grid data array
+   * @return {array} - return an array of array with value as an object in this format : { value : ... }
+   */
+  function formatExportRows(rows) {
+    return (rows || []).map(row => {
+      const code = row.inventory && row.inventory.code ? row.inventory.code : null;
+      const description = row.inventory && row.inventory.text ? row.inventory.text : null;
+      const lot = row.lot && row.lot.label ? row.lot.label : null;
+      const price = row.inventory && row.inventory.unit_cost ? row.inventory.unit_cost : null;
```

Comment:
I still feel like we could do much better than this, but I'll have to take a look at the code to see.